src/main/java/com/example/BookService/Service/BookService.java

```diff
@@ -0,0 +1,84 @@
+package com.example.BookService.Service;
+
+import com.example.BookService.DTO.BookDTO;
+import com.example.BookService.Entity.Book;
+import com.example.BookService.Repository.BookRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class BookService {
+
+    private final BookRepository bookRepository;
+    private final WebClient webClient;
+    public BookService(BookRepository bookRepository, WebClient.Builder webClientBuilder) {
+        this.bookRepository = bookRepository;
+        this.webClient=webClientBuilder.baseUrl("http://order-service:8080").build();
+    }
+
+    private BookDTO toDTO(Book book){
+        return new BookDTO(book.getId(), book.getTitle(), book.getAuthor(), book.getPrice(), book.getStock());
+    }
+
+    private Book toEntity(BookDTO bookDTO){
+        return new Book(bookDTO.getId(), bookDTO.getTitle(), bookDTO.getAuthor(), bookDTO.getPrice(), bookDTO.getStock());
+    }
+
+    public List<BookDTO> getAllBooks() {
+        return bookRepository.findAll().stream()
+                .map(this::toDTO).collect(Collectors.toList());
+    }
+
+    public Optional<BookDTO> getBookById(Long id) {
+        return bookRepository.findById(id).map(this::toDTO);
+    }
+
+    public BookDTO createBook(BookDTO bookDTO) {
+        Book book = toEntity(bookDTO);
```

Comment:
Would be nice to have more strict validations: for example the price and stock should not be less than 0. You can minimum number of chars for title and author.